src/v/storage/disk_log_impl.cc

```diff
@@ -2584,7 +2584,37 @@ disk_log_impl::get_reclaimable_offsets(gc_config cfg) {
           human::bytes(point.size));
     }
 
+    /*
+     * cache this for access by the health
+     */
+    _reclaimable_local_size_bytes = std::accumulate(
+      res.effective_local_retention.cbegin(),
+      res.effective_local_retention.cend(),
+      size_t{0},
+      [](size_t acc, reclaimable_offsets::offset update) {
+          return acc + update.size;
+      });
+
```

Comment:
I think this makes sense to cache, but I wonder how it will interact with local GC. Could we end up in a scenario where `size_bytes < reclaimable_size_bytes` in the report?

I'm wondering if we could end up with some scenario like:
- size_bytes reported to be 100
- reclaimable_local_size_bytes is 30 (local retention wants us to keep 70). this gets cached
- GC runs and size_bytes is now 70, but reclaimable_local_size_bytes is still 30
- it now looks like the partition is effectively size 40? (not sure if this is actually what the partition balancer will do)

Perhaps we should invalidate `_reclaimable_local_size_bytes` if we truncate to the local retention point, or when `size_bytes` is changed?

Reply:
>Could we end up in a scenario where size_bytes < reclaimable_size_bytes in the report?

it wouldn't surprise me. IIRC we have a similar problem in health report w.r.t. disk space statistics. like one source is the OS vfstat and the other is our always-kind-of-out-of-date tracking that we do. In this case the balancer normalizes anything in some sane way.

>Perhaps we should invalidate _reclaimable_local_size_bytes if we truncate to the local retention point, or when size_bytes is changed?

this seems like a reasonable thing to do to get fresher data. ill add a step to refresh the cached value after a prefix truncation event?

Reply:
addressed in the latest force push